Sources/OpenTelemetry/Tracing/Processing/Batch/OTelBatchSpanProcessorConfiguration.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2023 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// The configuration options for an ``OTelBatchSpanProcessor``.
+public struct OTelBatchSpanProcessorConfiguration: Sendable {
+    /// The maximum queue size.
+    ///
+    /// - Warning: After this size is reached spans will be dropped.
+    public var maximumQueueSize: UInt
+
+    /// The maximum delay between two consecutive span exports.
+    public var scheduleDelay: Duration
+
+    /// The maximum batch size of each export.
+    ///
+    /// - Note: If the queue reaches this size, a batch will be exported even if ``scheduleDelay`` has not elapsed.
+    public var maximumExportBatchSize: UInt
+
+    /// The duration a single export can run until it is cancelled.
+    public var exportTimeout: Duration
+
+    /// Create a batch span processor configuration.
+    ///
+    /// - Parameters:
+    ///   - environment: The environment variables.
+    ///   - maximumQueueSize: A maximum queue size used even if `OTEL_BSP_MAX_QUEUE_SIZE` is set. Defaults to `2048` if both are `nil`.
+    ///   - scheduleDelay: A schedule delay used even if `OTEL_BSP_SCHEDULE_DELAY` is set. Defaults to `5` seconds if both are `nil`.
+    ///   - maximumExportBatchSize: A maximum export batch size used even if `OTEL_BSP_MAX_EXPORT_BATCH_SIZE` is set. Defaults to `512` if both are `nil`.
+    ///   - exportTimeout: An export timeout used even if `OTEL_BSP_EXPORT_TIMEOUT` is set. Defaults to `30` seconds if both are `nil`.
+    public init(
+        environment: OTelEnvironment,
+        maximumQueueSize: UInt? = nil,
+        scheduleDelay: Duration? = nil,
+        maximumExportBatchSize: UInt? = nil,
+        exportTimeout: Duration? = nil
+    ) {
+        self.maximumQueueSize = environment.requiredValue(
```

Comment:
Okay last comment about the env stuff. I find it super weird to have those in all of the types. I think we should start simple with a value type for the configuration of each piece and then maybe provide convenience inits that read from well known env keys.

Reply:
I'd really like to keep the environment variable checks in the client since they are well defined and it's expected that a client handles these: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md

I don't know how with convenience inits we would deal with the fact that for some of the arguments a user may wants to provide a value via Swift whereas for others they want to use the default value, optionally overriding it via the well-known ENV variables.